src/lib/seam/components/DeviceTable/DeviceRow.tsx

```diff
@@ -16,6 +17,48 @@ interface Props {
   onBack?: () => void
 }
 
+interface DeviceStatusesProps {
+  device: UseDevicesData[number]
+}
+
+function accountIsOffline(device: UseDevicesData[number]): boolean {
+  return (
+    device.errors.filter((error) => 'is_connected_account_error' in error)
+      .length > 0
+  )
+}
+
+function deviceIsOffline(device: UseDevicesData[number]): boolean {
+  return (
+    device.errors.filter((error) => error.error_code === 'device_disconnected')
+      .length > 0
+  )
+}
+
+function DeviceStatuses({ device }: DeviceStatusesProps): JSX.Element | null {
+  if (accountIsOffline(device)) {
+    return (
+      <div className='seam-device-statuses'>
+        <AccountOnlineStatus device={device} />
+      </div>
+    )
+  }
+  if (deviceIsOffline(device)) {
+    return (
+      <div className='seam-device-statuses'>
+        <DeviceOnlineStatus device={device} />
+      </div>
+    )
+  }
+  return (
+    <div className='seam-device-statuses'>
+      <DeviceOnlineStatus device={device} />
+      <BatteryStatus device={device} />
+      <LockStatus device={device} />
+    </div>
+  )
+}
```

Comment:
Few general style note for this project that we can't really lint against yet. (1) For non-component functions we prefer arrow functions and (2) bool consts and bool functions should be named with the `is`, etc., in front, e.g., `const isAccountOffline = ` and (3) these internal methods and components should be ordered after they are used, so they should all be below the main export, ref https://stackoverflow.com/questions/1760850/best-practice-ordering-of-public-protected-private-within-the-class-definition/1760877#1760877